extract/extractor.go

```diff
@@ -67,17 +68,37 @@ func (e *Extractor) processItems(cf read.Contentful, typeResult mapper.TypeResul
 		log.Fatal(err)
 	}
 
-	writer := write.Writer{Store: e.Store}
+	archetypeDataMap := make(map[string]map[string]interface{})
+	reader := read.Reader{Store: e.RStore}
+	writer := write.Writer{Store: e.WStore}
 	tc := translate.TranslationContext{Result: itemResult, TransConfig: e.TransConfig}
 	for _, item := range itemResult.Items {
-
-		itemType, err := typeResult.GetType(item.ContentType())
+		contentType := item.ContentType()
+		itemType, err := typeResult.GetType(contentType)
 		if err != nil {
-			log.Fatal(err)
+			log.Fatalln(err)
+		}
+
+		if archetypeDataMap[contentType] == nil {
```

Comment:
Might be at the point here, complexity wise, where this needs to be busted out to it's own function